AIO/src/org/aio/activities/quests/TheRestlessGhost.java

```diff
@@ -0,0 +1,190 @@
+package org.aio.activities.quests;
+
+import org.aio.activities.activity.Activity;
+import org.aio.activities.banking.DepositAllBanking;
+import org.aio.util.Sleep;
+import org.osbot.rs07.api.map.Area;
+import org.osbot.rs07.api.model.NPC;
+import org.osbot.rs07.api.model.RS2Object;
+import org.osbot.rs07.api.ui.EquipmentSlot;
+import org.osbot.rs07.api.ui.Tab;
+
+import java.util.stream.Stream;
+
+public class TheRestlessGhost extends QuestActivity {
+
+    private static final Area CHURCH = new Area(3240, 3204, 3247, 3215);
+    private static final Area URHNEY = new Area(3144, 3173, 3151, 3177);
+    private static final Area COFFIN = new Area(3247, 3195, 3252, 3190);
+    private static final Area WIZARD = new Area(3116, 9564, 3121, 9569);
+
+    private static final String[] AERECT_OPTIONS = {
+            "I'm looking for a quest!",
+            "Ok, let me help then."
+    };
+
+    private static final String[] URHNEY_OPTIONS = {
+            "Father Aereck sent me to talk to you.",
+            "He's got a ghost haunting his graveyard."
+    };
+
+    private static final String[] GHOST_OPTIONS = {
+            "Yep, now tell me what the problem is."
+    };
+
+    private static final int INVENTORY_SLOTS_REQUIRED = 4;
+
+    private static final String[] ITEMS_NEEDED = {
+            "Ghost's skull",
+            "Ghostspeak amulet",
+    };
+
+    private final DepositAllBanking depositAllBanking = new DepositAllBanking(ITEMS_NEEDED);
+
+    public TheRestlessGhost() {
+        super(Quest.THE_RESTLESS_GHOST);
+    }
+
+    @Override
+    public void onStart() {
+        depositAllBanking.exchangeContext(getBot());
+    }
+
+    @Override
+    public void runActivity() throws InterruptedException {
+        if (!getInventory().contains(ITEMS_NEEDED) && getInventory().getEmptySlotCount() < INVENTORY_SLOTS_REQUIRED) {
+            depositAllBanking.run();
+        } else if (getTabs().getOpen() != Tab.INVENTORY) {
+            getTabs().open(Tab.INVENTORY);
+        } else {
+            switch (getProgress()) {
+                case 0:
+                    talkToAereck();
+                    break;
+                case 1:
+                    talkToUrhney();
+                    break;
+                case 2:
+                    talkToGhost();
+                    break;
+                case 3:
+                    searchAlter();
+                    break;
+                case 4:
+                    placeSkull();
+                    break;
+                case 5:
+                    // Make sure we are not in the cut scene
+                    if (getWidgets().get(277, 2).isVisible()) {
```

Comment:
We should try not to use static widget IDs, they have a tendency to change and the script will break. What widget is this? Is there any other way of identifying the widget?